### DISCUSSION TITLE
gpio delays

### DISCUSSION THREAD

**Discussion by thebadboyltz** [technyon/nuki_hub#539] [Q&A] [answered]
Hello,
is it possible to modify the GPIO delay, I use a reed switch on my front door currently connected to an esp32 which goes back to HA.
When I close my door after 5 seconds my lock closes.

I want to connect my reed switch directly to gpio 2 of the NuKiHub but I would like to set a delay of 5000ms so that the door closes after 5 seconds after gpio2 has become low.

thank you

**Comment by technyon** [comment, marked as answer]
No this is not possible with Huki Hub alone. What you could do is to configure the pin as general input, and write a script that reacts to input of pin 2. It should delay the reaction 5 seconds and send a command to lock the door back to Huki Hub.

**Comment by thebadboyltz** [comment]
thanks for the reply
